Scripting-Engine/Scripting-Engine/LeagueSandbox-Default/Champions/Ezreal/E.cs

```diff
@@ -22,54 +23,44 @@ static void onFinishCasting(Champion owner, Spell spell)
             Vector2 trueCoords;
 
             if (to.Length() > 475) {
-                //to = to.Normalize()
-                //local range = to * 475
-                //trueCoords = current:copy() + range
-            } else {
-                trueCoords = new Vector2(spell.X, spell.Y);
-             }
-            /*
-
-    if to:length() > 475 then
-        to = to:normalize()
-                local range = to * 475
-                trueCoords = current:copy() + range
+                to = Vector2.Normalize(to);
+                Vector2 range = to * 475;
+                trueCoords = current + range;
+            }
             else
-        trueCoords = Vector2:new(spell.X, spell.Y)
-            end
-        
-    addParticle(owner, "Ezreal_arcaneshift_cas.troy", owner.X, owner.Y);
-            teleportTo(owner, trueCoords.x, trueCoords.y)
-            addParticleTarget(owner, "Ezreal_arcaneshift_flash.troy", owner);
-
-            local target = nil
-            local units = getUnitsInRange(owner, 700, true)
-        
+            {
+                trueCoords = new Vector2(spell.X, spell.Y);
+            }
+            ApiFunctionManager.AddParticle(owner, "Ezreal_arcaneshift_cas.troy", owner.X, owner.Y);
+            spell.Teleport(owner, trueCoords.X, trueCoords.Y);
+            ApiFunctionManager.AddParticleTarget(owner, "Ezreal_arcaneshift_flash.troy", owner );
+            Unit target = null;
+            List<Unit> units = ApiFunctionManager.GetUnitsInRange(owner, 700, true);
 
-    for i = 0, units.Count - 1 do
-                    value = units[i]
-     
-             local distance = 700
-                if owner.Team ~= value.Team then
-                    if Vector2:new(trueCoords.x, trueCoords.y):distance(Vector2: new(value.X, value.Y)) <= distance then
-                 target = value
-                        distance = Vector2:new(trueCoords.x, trueCoords.y):distance(Vector2: new(value.X, value.Y))
-                    end
-        end
-            end
-            if (target) and(not unitIsTurret(target)) then
-               addProjectileTarget("EzrealArcaneShiftMissile", target)
-            end
-            */
+            for(int i = 0; i < units.Count; i++)
+            {
+                Unit value = units[i];
+                float distance = 700;
+                if(owner.Team != value.Team)
+                {
+                    if(Vector2.Distance(new Vector2(trueCoords.X, trueCoords.Y), new Vector2(value.X, value.Y)) <= distance)
+                    {
+                        target = value;
+                        distance = Vector2.Distance(new Vector2(trueCoords.X, trueCoords.Y), new Vector2(value.X, value.Y));
+                    }
+                }
+            }
+            if(!ApiFunctionManager.UnitIsTurret(target))
```

Comment:
Why the fuck is this method even a thing. Abstract this instead by keeping track if `Unit`s are buildings or entities. Something like

    enum UnitKind {
        ENTITY,  // any visible entity with hp/movement, think minions/champs
        BUILDING, // buildings (towers, inhib, nexus)
        PROJECTILE, // projectiles
        STATIC // static units (jarvan ult terrain, yasuo windwall)
    }